aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -33,6 +35,74 @@ using namespace at::sparse_csr;
 // certain utiliy functions are usable from sparse COO.
 using namespace at::sparse;
 
+Tensor& addmm_out_sparse_csr_dense_cuda(
+  const Tensor& self,
+  const SparseCsrTensor& sparse,
+  const Tensor& dense,
+  const Scalar& beta,
+  const Scalar& alpha,
+  Tensor& r)
+{
+
+  TORCH_INTERNAL_ASSERT(sparse.is_sparse_csr());
+  Tensor t = *expand_size(self, {sparse.size(0), dense.size(1)}, "addmm_out_sparse_csr");
+
+  TORCH_INTERNAL_ASSERT(t.device().type() == kCUDA);
+  TORCH_CHECK(
+      r.device().type() == kCUDA,
```

Comment:
why are you checking only device types and not exact devices? Also, likely the device checks are not needed at all because they are now automatically generated (but you should make sure that this is tested)

Reply:
You are right, I tested it and these checks are automatically generated thanks to https://github.com/pytorch/pytorch/pull/56872. I will remove them as they are redundant.

Reply:
Update: check are automatically generated but is not always called. I think this should be addressed in a different PR as it not only related to CSR tensors but COO tensors as well. 
I detailed the issue here: https://github.com/pytorch/pytorch/issues/59058